### ISSUE TITLE
Holopad Megaissue

### ISSUE THREAD

**Issue by ElectroJr** [space-wizards/space-station-14#33986]
Megaissue for any holopad bugs/issues that need to be fixed. This is related to #33954. Maybe the issues should be merged, but I wanted to separate issues & bugs from future features.

- [ ] Holopad networking needs to be reworked
	- [ ] Holopads are client-authoritative. Modified clients can make their holopads appear however they want.
	- [ ] Holopads likely don't play nicely with replays, the appearance data should be part of a component state, not a networked event.
	- The easiest fix for this is likely to use PVS overrides to directly send the viewed entity and their children, and then having the client copy the data. If we eventually get sprite nesting (using an entity as a layer in another entity's sprite), it should use that.

**Comment by Errant-4** [comment]
Holopads absolutely spam the chat of observers/aghosts when they are broadcasting, because you hear the speecs of _every last one of them_

**Comment by chromiumboy** [comment]
I actually originally used PVS overrides for this, and it worked just fine, but I got worried about how every client was getting holopad users data and that cheaters might exploit this. But if you don't think it will be problematic, I'll make the switch back

**Comment by ElectroJr** [comment]
> I actually originally used PVS overrides for this, and it worked just fine, but I got worried about how every client was getting holopad users data and that cheaters might exploit this. But if you don't think it will be problematic, I'll make the switch back

Yeah, It's also not a great solution, and also kind of sucks for performance, but having it be client authoritative feels pretty icky as well. The same problem more or less also exists for the round end lobby (#29160). But I also don't really know of a good way to fix either one aside from just sending the entities.

**Comment by TGRCdev** [comment]
Holopads don't work with displacement maps, which cause vox to be warped into human proportions.

![image](https://github.com/user-attachments/assets/1881bf1e-64f1-4068-bdc6-99792d251398)

**Comment by chromiumboy** [comment]
> > I actually originally used PVS overrides for this, and it worked just fine, but I got worried about how every client was getting holopad users data and that cheaters might exploit this. But if you don't think it will be problematic, I'll make the switch back
> 
> Yeah, It's also not a great solution, and also kind of sucks for performance, but having it be client authoritative feels pretty icky as well. The same problem more or less also exists for the round end lobby (#29160). But I also don't really know of a good way to fix either one aside from just sending the entities.

At least we'd only be adding a much smaller number of players to the PVS at a time, up to two per call in progress. Even if multiple people speak into a the same holopad, the holopad on the other end only displays one person at a time, so one one will be added to the PVS override. It'll also fix issues like the displacement maps cos I don't have worry about applying all the sprites by hand

**Comment by LordEclipse** [comment]
If an A.I. is using a holopad and its unanchored, it breaks the A.I. eye, which can only be fixed by the A.I. returning to core.

**Comment by slarticodefast** [comment]
Newly spawned holopads do not show the coundown for the broadcast option when it is running. The button does not work until the cooldown is over, so this is visual only.

**Comment by ilovehans10** [comment]
Another thing that might be intended but feels wrong is that mute players are absolutely fucked if they have a PAI or anything else that talks near them. This feels like it could be intentional, but at the same time holopads feel like the ultimate accessability option for communication. A simple fix for this would be to have a "Gain Focus" button in the holopads UI when on a call, so if my PAI makes a glib remark it won't prevent me from being able to gesture over the call.

Another thing to build on this idea would be to have a "Lock Focus" option that requires extra ID access. That way if I'm mute and plan on using holopads I can go to HOP round start and give them my Holopad Focus Lock request form, and no longer have to worry about a mouse screaming near me. Also department heads should probably have this access too as being CE and having a Technical Assistant budge into a holocall can be undesired.

**Comment by chromiumboy** [comment]
The focusing issue with PAIs can be partly solved by having it so that when a speaking player is in the inventory of another the holopad focuses on the parent player

The second part is to have emoting bring the focus back to the player in adddtion to speaking. Then there would be no need to have focus buttons
